### ISSUE TITLE
Import comparer may fail in rare edge cases

### ISSUE THREAD

**Issue by MikeStower** [QAAFI/REMS2020#60]
The comparer that the importer uses to determine if data already exists uses a series of bitwise XOR operations to generate the hash and determine equality. However, this operation is commutative and acts upon multiple small range integer values, which means there are rare situations where distinct elements could generate the same hash.

For example, the SoilLayerComparer uses both TraitId and PlotId in its hash function. There may be a situation where one SoilLayerData has a TraitId equal to the PlotId of the one being compared, and vice versa.

1) `a.TraitId == b.PlotId`
2)  `a.PlotId == b.TraitId;`
3) `a.TraitId ^ a.PlotId == b.TraitId ^ b.PlotId`

In other words, 1) and 2) imply 3). Even though `a` and `b` are different objects, the comparer would produce the same hash code.

Though this is extremely unlikely, (as far as I can tell, it didn't occur with any of the test data), it should probably still be fixed at some point.

**Comment by MikeStower** [comment]
It might be fixable by multiplying each hash element by a distinct prime number.
